lib/async/http/protocol/http1/response.rb

```diff
@@ -11,17 +11,21 @@ module Protocol
 			module HTTP1
 				class Response < Protocol::Response
 					def self.read(connection, request)
-						if parts = connection.read_response(request.method)
-							self.new(connection, *parts)
+						while parts = connection.read_response(request.method)
+							response = self.new(connection, *parts)
+							
+							if response.final?
+								return response
+							end
 						end
 					end
 					
 					UPGRADE = 'upgrade'
-
-          # @attribute [String] The HTTP response line reason.
+					
+					# @attribute [String] The HTTP response line reason.
 					attr :reason
-
-					# @parameter reason [String] HTTP response line reason phrase
+					
+					# @parameter reason [String] HTTP response line reason phrase.
 					def initialize(connection, version, status, reason, headers, body)
```

Comment:
Looks here like the docs for the initialize args aren't fully here, or perhaps I am missing something that ain't visible right here in the commit diff.

Reply:
You are right, the documentation could do with some improvement I suppose.